sootup.core/src/main/java/sootup/core/cache/MutableCache.java

```diff
@@ -0,0 +1,9 @@
+package sootup.core.cache;
+
+import sootup.core.model.SootClass;
+import sootup.core.types.ClassType;
+
+/** Interface for caches which are mutable and allow classes to be removed from. */
+public interface MutableCache<S extends SootClass<?>> extends Cache<S> {
+  S removeClass(ClassType classType);
```

Comment:
```suggestion
  default S modifyClass(@Nonnull ClassType classType){
  
  removeClass(classType);
  addClass(classType);
  }
```
as a convenience method?

Reply:
Do you mean a replaceClass method, which removes an old class and instead adds a new one?
The method you are suggesting adds and removes the same class.

Reply:
sure.. someone was excited of using the suggestion feature ;)